aten/src/ATen/native/UpSample.cpp

```diff
@@ -30,6 +30,29 @@ TORCH_API c10::SmallVector<int64_t, 3> compute_output_size(
   TORCH_CHECK(false, "Must specify exactly one of output_size and scale_factors");
 }
 
+TORCH_API c10::SmallVector<SymInt, 3> compute_output_size(
+    c10::SymIntArrayRef input_size,  // Full input tensor size.
+    at::OptionalIntArrayRef output_size,
+    c10::optional<c10::ArrayRef<double>> scale_factors) {
+  const auto spatial_dimensions = static_cast<int64_t>(input_size.size()) - 2;
+  if (output_size) {
+    TORCH_CHECK(!scale_factors, "Must specify exactly one of output_size and scale_factors");
+    TORCH_CHECK(static_cast<int64_t>(output_size->size()) == spatial_dimensions);
+    return {output_size->data(), output_size->data() + output_size->size()};
+  }
+  if (scale_factors) {
+    TORCH_CHECK(!output_size, "Must specify exactly one of output_size and scale_factors");
+    TORCH_CHECK(static_cast<int64_t>(scale_factors->size()) == spatial_dimensions);
+    c10::SmallVector<SymInt, 3> ret;
+    for (const auto i : c10::irange(spatial_dimensions)) {
+      const auto scaled_float = SymFloat(input_size[i+2]) * scale_factors.value()[i];
+      ret.push_back(scaled_float.floor());
+    }
+    return ret;
+  }
+  TORCH_CHECK(false, "Must specify exactly one of output_size and scale_factors");
```

Comment:
We have an implementation of this in Python at torch/_decomp/decompositions.py:def upsample_compute_output_size(input_size, output_size, scale_factors) for the upsample decomps, so it feels like it would be nicer if the aa composite was ported to Python, like the other decomps are, then you wouldn't need to dupe this function.